src/mysql/types.cr

```diff
@@ -154,9 +158,71 @@ abstract struct MySql::Type
     end
   end
   decl_type LongLong, 0x08u8, ::Int64
-  decl_type Int24, 0x09u8
-  decl_type Date, 0x0au8
-  decl_type Time, 0x0bu8
+  decl_type Int24, 0x09u8, ::Int32
+
+  def self.datetime_read(packet)
```

Comment:
isn't this extracted from `DateTime.read`? Then it should be used there to avoid code duplication.